tests/epyccel/recognised_functions/test_math_funcs.py

```diff
@@ -377,12 +382,927 @@ def floor_return_type_real(x):
+        pytest.param("c", marks = pytest.mark.c),
+        pytest.param("fortran", marks = [
+            pytest.mark.xfail(reason="log2 not implemented"),
+            pytest.mark.fortran]
+        )
+    )
+)
+def test_log2_phrase(language):
+    @types('real','real')
+    def log2_phrase(x,y):
+        from math import log2
+        a = log2(x)+log2(y)
+        return a
+
+    f2 = epyccel(log2_phrase, language = language)
+    low = min_float
+    high = max_float
+    x = uniform(low=low, high=high)
+    y = uniform(low=low, high=high)
+    assert(isclose(f2(x,y) ,  log2_phrase(x,y), rtol=1e-15, atol=1e-15))
+
+#--------------------------- log10 function ------------------------------#
+
+def test_log10_call(language):
+    @types('real')
+    def log10_call(x):
+        from math import log10
+        return log10(x)
+
+    f1 = epyccel(log10_call, language = language)
+    low = min_float
+    high = max_float
+    x = uniform(low=low, high=high)
+    assert(isclose(f1(x) ,  log10_call(x), rtol=1e-15, atol=1e-15))
+    assert isinstance(f1(x), type(log10_call(x)))
+
+def test_log10_phrase(language):
+    @types('real','real')
+    def log10_phrase(x,y):
+        from math import log10
+        a = log10(x)+log10(y)
+        return a
+
+    f2 = epyccel(log10_phrase, language = language)
+    low = min_float
+    high = max_float
+    x = uniform(low=low, high=high)
+    y = uniform(low=low, high=high)
+    assert(isclose(f2(x,y) ,  log10_phrase(x,y), rtol=1e-15, atol=1e-15))
+
+#--------------------------------- Pow function ------------------------------#
+@pytest.mark.parametrize( 'language', (
+        pytest.param("c", marks = pytest.mark.c),
+        pytest.param("fortran", marks = [
+            pytest.mark.xfail(reason="pow not implemented"),
+            pytest.mark.fortran]
+        )
+    )
+)
+def test_pow_call(language):
+    @types('real', 'real')
+    def pow_call(x, y):
+        from math import pow as my_pow
+        return my_pow(x, y)
+
+    f1 = epyccel(pow_call, language = language)
+    high = 10
+    # case 1: x > 0
+    x = uniform(low=min_float)
+    y = uniform(low=-high, high=high)
+    assert(isclose(f1(x, y) , pow_call(x, y), rtol=1e-15, atol=1e-15))
+    assert isinstance(f1(x, y), type(pow_call(x, y)))
+
+    # case 2: x = 0 and y > 0
+    x = 0.0
+    y = uniform(high=high)
+    assert(isclose(f1(x, y), pow_call(x, y), rtol=1e-15, atol=1e-15))
+
+    # case 3: x < 0 and y is integer
+    x = uniform(low=-high, high=0)
+    y = randint(high)
+    assert(isclose(f1(x, y), pow_call(x, y), rtol=1e-15, atol=1e-15))
+
+#------------------------------- Hypot function ------------------------------#
+@pytest.mark.parametrize( 'language', (
+        pytest.param("c", marks = pytest.mark.c),
+        pytest.param("fortran", marks = [
+            pytest.mark.xfail(reason="hypot not implemented"),
+            pytest.mark.fortran]
+        )
+    )
+)
+def test_hypot_call(language):
```

Comment:
This test is xpassing for fortran

Reply:
yes, you are right.
I am handling that, and I will add more functions that are already in fortran.